### PR TITLE
update for v0.6.0 release

### PR DESCRIPTION
**Pull request by desimone**
See: 
- https://github.com/pomerium/pomerium/releases/tag/v0.6.0
- https://github.com/pomerium/pomerium/pull/464

Signed-off-by: Bobby DeSimone <bobbydesimone@gmail.com>

### CODE REVIEW

**Comment by desimone** [comment]
@travisgroth less a request for review and more of a cry for help :)

File: charts/pomerium/templates/tls-secrets.yaml [hunk 1]
```diff
@@ -143,6 +150,12 @@ data:
   {{ template "pomerium.authorize.tlsSecret.certName" . }}: {{ .Values.authorize.tls.cert | b64enc }}
   {{ template "pomerium.authorize.tlsSecret.keyName" . }}: {{ .Values.authorize.tls.key | b64enc }}
 {{-     end }}
+{{-     if and (not .Values.cache.existingTLSSecret) .Values.cache.tls.cert .Values.cache.tls.key }}
+---
+{{ template "pomerium.cache.tlsSecretObject" . }}
```
Comment (travisgroth) [inline charts/pomerium/templates/tls-secrets.yaml]: You'll need a block at the top of this file for the TLS secret hooks

File: charts/pomerium/templates/_helpers.tpl [hunk 2]
```diff
@@ -148,6 +167,23 @@ Adapted from : https://github.com/helm/charts/blob/master/stable/drone/templates
 {{- end -}}
 {{- end -}}
 
+{{/* Determine secret name for cache TLS Cert */}}
+{{- define "pomerium.cache.tlsSecret.name" -}}
+{{- if .Values.cache.existingTLSSecret -}}
+{{- .Values.cache.existingTLSSecret | trunc 63 | trimSuffix "-" -}}
+{{- /* TODO in future: Remove legacy logic */ -}}
```
Comment (travisgroth) [inline charts/pomerium/templates/_helpers.tpl]: I think we should avoid carrying any legacy TODOs on the cache service at this point.  My thought was to remove the rest with the operator release.  No reason to add more debt ahead of that.

File: charts/pomerium/templates/ingress.yaml [hunk 3]
```diff
@@ -17,6 +17,7 @@ spec:
     - secretName: {{ default .Values.ingress.secretName .Values.ingress.secret.name}}
       hosts:
         - {{ printf "authorize.%s" .Values.config.rootDomain | quote }}
+        - {{ printf "cache.%s" .Values.config.rootDomain | quote }}
```
[the remark this answers is not in the document]
Comment (desimone) [reply charts/pomerium/templates/ingress.yaml]: 🤦‍♂

File: charts/pomerium/templates/_helpers.tpl (hunk 2 above)
Comment (desimone) [reply charts/pomerium/templates/_helpers.tpl]: 👍

File: charts/pomerium/templates/authenticate-deployment.yaml [hunk 4]
```diff
@@ -66,7 +66,9 @@ spec:
         - name: SERVICES
           value: authenticate
         - name: AUTHENTICATE_SERVICE_URL
-          value: {{ default (printf "https://authenticate.%s" .Values.config.rootDomain ) .Values.proxy.authenticateServiceUrl }}
+          value: {{ default (printf "https://authenticate.%s" .Values.config.rootDomain ) .Values.proxy.cacheServiceUrl }}
+        - name: CACHE_SERVICE_URL
+          value: {{ default (printf "https://%s.%s.svc.cluster.local" (include "pomerium.cache.fullname" .) .Release.Namespace ) .Values.proxy.cacheInternalUrl}}
```
[the remark this answers is not in the document]
Comment (desimone) [reply charts/pomerium/templates/authenticate-deployment.yaml]: Good catch 🤦‍♂

File: charts/pomerium/templates/tls-secrets.yaml (hunk 1 above)
Comment (desimone) [reply charts/pomerium/templates/tls-secrets.yaml]: Added

File: charts/pomerium/templates/authenticate-deployment.yaml [hunk 5]
```diff
@@ -66,7 +66,9 @@ spec:
         - name: SERVICES
           value: authenticate
         - name: AUTHENTICATE_SERVICE_URL
-          value: {{ default (printf "https://authenticate.%s" .Values.config.rootDomain ) .Values.proxy.authenticateServiceUrl }}
+          value: {{ default (printf "https://authenticate.%s" .Values.config.rootDomain ) .Values.proxy.cacheServiceUrl }}
```
[the remark this answers is not in the document]
Comment (desimone) [reply charts/pomerium/templates/authenticate-deployment.yaml]: 🤦‍♂

File: charts/pomerium/README.md [hunk 6]
```diff
@@ -25,7 +25,7 @@
 ## TL;DR;
 
 ```console
-helm install --name my-release stable/pomerium
+helm install my-release pomerium/pomerium
```
Comment (travisgroth) [inline charts/pomerium/README.md]: 😆

File: charts/pomerium/README.md [hunk 7]
```diff
@@ -176,6 +184,13 @@ A full listing of Pomerium's configuration variables can be found on the [config
 
 ## Upgrading
 
+### 5.0.0
+
+- A new service, cache, was added to this chart release.
```
Comment (travisgroth) [inline charts/pomerium/README.md]: 👍

File: charts/pomerium/README.md (hunk 6 above)
Comment (desimone) [reply charts/pomerium/README.md]: I feel provoked. 😆
